source/roadmap.md

```diff
@@ -147,6 +147,22 @@ Desired features include:
 
 **Expected outcomes:** fpm can generate source code distributions of packages that can be build using CMake and/or Meson.
 
+## Improving Compilers
+
+The aim of this project is to create and maintain an open source Fortran compiler as a community, here is a non-exhaustive list of (independent) features:
+
+* compile all Fortran projects to binaries
+* work interactively (Jupyter notebooks)
+* in the browser
+* works as the foundation for linters, VSCode language integration and documentation generators
+* runs well on modern hardware (CPU, GPU, ...)
+
+There are three open source compilers under development:
+
+* LFortran: https://lfortran.org/
+* Flang: https://flang.llvm.org/docs/
+* GFortran: https://gcc.gnu.org/wiki/GFortran
+
```

Comment:
I would consider adding the actual projects and deliverables we have in mind for these (at least for LFortran which you control):

https://github.com/fortran-lang/fortran-lang.org/wiki/GSoC-2022-Project-ideas#language-server
https://github.com/fortran-lang/fortran-lang.org/wiki/GSoC-2022-Project-ideas#parser--asr-for-f2py

For the LSP stuff I would add some summaries from the discussion with the vscode-cpptools devs and email conversations. I will dig them up at a later point

```suggestion

### LFortran: Parser + ASR (for F2PY)

Enhance the frontend parser to allow for more generic extensions and keywords. In particular, parsing the `pyf` or signature file format consumed by F2PY. This would mean that SciPy can be directly transformed to the ASR representation without a fixed form parser.

This also involves adding nodes to the ASR itself.

**Expected outcomes:** LFortran's ASR is improved such that it can parse the `pyf` format consumed by F2PY.

### LFortran: Language Server

This project would be used to first serialize the ASR and then use it within a language server.

**Expected outcomes:** LFortran can be used as a Fortran language server that can be used in other software such as source code editors and IDEs.

```

Reply:
@certik wanna have a look and tell me if I forgot something?

Reply:
This requires a design discussion.

I would not recommend parsing the f2py output, because it means supporting that syntax, which complicates maintenance. For f2py's application, I think it's better to just parse Fortran itself, and possibly annotate with comments, both of which we can parse.

Also adding ASR nodes must be done carefully. We try avoiding adding nodes which are only used by one tool, as opposed to all of Fortran (or other languages).

Reply:
Let me try to write down a roadmap, for the next half a year, which effectively is to compile "fpm".